Add optional bold to theme colors

pdf opened the pull request:
Adds an optional field to color definitions, that if set to `'bold'` will enable
bold foreground text.

pdf committed:
Add optional attributes to theme colors

Adds an optional field to color definitions, that may be set to a
comma-separated string containing a combination of the following
attributes:

- bright (alias: bold)
- dim
- standout (alias: italic)
- underscore (alias: underline)
- blink
- reverse
- hidden
- strikeout (alias: strikethrough)

When set, the corresponding modifier will be added to the color, to
alter text display on supported terminals.

pdf commented:
I left the powerlineclone theme update in a separate commit in case you don't want to take it.

edkolev commented:
Supporting boldness is a good feature. However, I suspect supporting boldness may result in issues, due to the powerline symbols - terminals don't like displaying these special symbols with bold. So even if we end up supporting boldness, I don't think the default themes should use it (but it will be easy for users to use bold if they would like to).

Terminals typically support setting foreground, background and attributes. Bold is just one of the possible attributes, there are others like `reverse`, `dim`, `blink`, `underscore` http://wiki.bash-hackers.org/scripting/terminalcodes

Could you modify the PR to handle any attribute (not just bold)? For example, the themes may set `[ 1, 2, "bold,underline" ]`, which would propagate to the prompt as the bold and underline escape codes

pdf commented:
Hello from the future :wink: 

I have updated the PR to include the following attributes:
- `bright` (alias: `bold`)
- `dim`
- `standout` (alias: `italic`)
- `underscore` (alias: `underline`)
- `blink`
- `reverse`
- `hidden`
- `strikeout` (alias: `strikethrough`)

And I've removed the commit modifying the promptlineclone theme as requested (though I have not observed any problem displaying bold separator symbols in a handful of terminals).

Supporting all the other attributes means having to reset them all individually, which is why I never got around to it originally, but I had to re-generate my prompt, so here it is.

thisishuey commented:
Any chance this can be merged? was looking for this functionality myself.

thisishuey commented:
@edkolev what else would be needed to get this PR merged in?
